tests/components/volumio/test_config_flow.py

```diff
@@ -0,0 +1,252 @@
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "TestVolumio"
+    assert result2["data"] == {**TEST_SYSTEM_INFO, **TEST_CONNECTION}
+
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_updates_unique_id(hass):
+    """Test a duplicate id aborts and updates existing entry."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id=TEST_SYSTEM_INFO["id"],
+        data={
+            "host": "dummy",
+            "port": 11,
+            "name": "dummy",
+            "id": TEST_SYSTEM_INFO["id"],
+        },
+    )
+
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    with patch(
+        "homeassistant.components.volumio.config_flow.Volumio.get_system_info",
+        return_value=TEST_SYSTEM_INFO,
+    ), patch("homeassistant.components.volumio.async_setup", return_value=True), patch(
+        "homeassistant.components.volumio.async_setup_entry", return_value=True,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], TEST_CONNECTION,
+        )
+
+    assert result2["type"] == "abort"
+    assert result2["reason"] == "already_configured"
+
+    assert entry.data == {**TEST_SYSTEM_INFO, **TEST_CONNECTION}
+
+
+async def test_empty_system_info(hass):
+    """Test old volumio versions with empty system info."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.volumio.config_flow.Volumio.get_system_info",
+        return_value={},
+    ), patch(
+        "homeassistant.components.volumio.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.volumio.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], TEST_CONNECTION,
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == TEST_CONNECTION["host"]
+    assert result2["data"] == {
+        "host": TEST_CONNECTION["host"],
+        "port": TEST_CONNECTION["port"],
+        "name": TEST_CONNECTION["host"],
+        "id": None,
+    }
+
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_cannot_connect(hass):
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.volumio.config_flow.validate_input",
```

Comment:
Please patch `get_system_info` to throw the side_effect `CannotConnectError ` instead here of patching the config flow

```
# pytest --cov=homeassistant/components/volumio/ --cov-report term-missing -- tests/components/volumio/test_*.py

----------- coverage: platform linux, python 3.8.1-final-0 -----------
Name                                              Stmts   Miss  Cover   Missing
-------------------------------------------------------------------------------
homeassistant/components/volumio/config_flow.py      68      2    97%   30-31
homeassistant/components/volumio/const.py             3      0   100%
-------------------------------------------------------------------------------
TOTAL                                                71      2    97%

```